### DISCUSSION TITLE
Paths of sudoless brew on Linux

### DISCUSSION THREAD

**Discussion by abitrolly** [Homebrew/discussions#119] [Everyday usage] [answered]
I am not a `brew` user. Just found https://docs.brew.sh/Homebrew-on-Linux

> Can install software to your home directory and so does not require sudo

Where does it download binaries and where it symlinks them when installed without `sudo`?

**Comment by gromgit** [comment]
From the page you pointed to, in the section after the one you quoted:
> The installation script installs Homebrew to `/home/linuxbrew/.linuxbrew` using `sudo` if possible and in your home directory at `~/.linuxbrew` otherwise.

**Comment by abitrolly** [comment]
@gromgit where it installs executables for other packages, and how do they appear in my path without `sudo`?

**Reply by jonchang** [reply]
You’d have to add Homebrew to your path yourself.

**Reply by abitrolly** [reply]
Ok. That will install only `brew` binary. What about other installed binaries?

Nevermind, everything goes into `$HOME/.linuxbrew/bin`.

**Comment by abitrolly** [comment, marked as answer]
Answering my own question. After installing, `brew` appears in `$HOME/.linuxbrew/bin` and not added to `$PATH`.
```shell
➜  ~ ls -la .linuxbrew/bin 
total 8
drwxrwxr-x.  2 anatoli anatoli 4096 Nov  8 14:27 .
drwxr-xr-x. 14 anatoli anatoli 4096 Nov  8 14:26 ..
lrwxrwxrwx.  1 anatoli anatoli   42 Nov  8 14:27 brew -> /home/anatoli/.linuxbrew/Homebrew/bin/brew
```

After installing `k9s` with `.linuxbrew/bin/brew install k9s`, the `.linuxbrew/bin` gets more symlinks.
```shell
➜  ~ ls -la .linuxbrew/bin          
total 8
drwxrwxr-x.  2 anatoli anatoli 4096 Nov  9 07:01 .
drwxr-xr-x. 14 anatoli anatoli 4096 Nov  8 14:26 ..
lrwxrwxrwx.  1 anatoli anatoli   42 Nov  8 14:27 brew -> /home/anatoli/.linuxbrew/Homebrew/bin/brew
lrwxrwxrwx.  1 anatoli anatoli   28 Nov  9 07:01 k9s -> ../Cellar/k9s/0.23.9/bin/k9s
-rw-r--r--.  1 anatoli anatoli    0 Nov  9 07:00 .keepme
lrwxrwxrwx.  1 anatoli anatoli   36 Nov  9 07:01 patchelf -> ../Cellar/patchelf/0.11/bin/patchelf
```

All paths related to `k9s` binary.
```shell
➜  ~ fd k9s -H                      
.cache/Homebrew/downloads/5f932f171a96027953b2736e24c3bb1b5d33164683da27bcb74f0984bedbc1ce--k9s-0.23.9.x86_64_linux.bottle.tar.gz
.cache/Homebrew/k9s--0.23.9.x86_64_linux.bottle.tar.gz
.linuxbrew/var/homebrew/linked/k9s
.linuxbrew/Cellar/k9s
.linuxbrew/opt/k9s
.linuxbrew/Cellar/k9s/0.23.9/.brew/k9s.rb
.linuxbrew/Cellar/k9s/0.23.9/bin/k9s
.linuxbrew/bin/k9s
```
